src/ng/parse.js

```diff
@@ -1788,6 +1797,7 @@ function $ParseProvider() {
             }
             cache[cacheKey] = parsedExpression;
           }
+          parsedExpression = expensiveChecksInterceptor(parsedExpression, expensiveChecks);
```

Comment:
Actually, that's what I meant by "move this inside the if-block". (Inside and _before_ `cache[cacheKey] = parsedExpression`.)
Thx for making it clear, @petebacondarwin :smiley: